### DISCUSSION TITLE
How do I render the same qmd multiple times, with different parameters, through _quarto.yml

### DISCUSSION THREAD

**Discussion by f2il-kieranmace** [quarto-dev/quarto-cli#7805] [Q&A] [answered]
### Description

I have one `qmd` and want to render it multiple times. How would I pass these render params to `_quarto.yml`?

using cli I would do the following: 

```sh
quarto render analysis.qmd -P USA
quarto render analysis.qmd -P UK
quarto render analysis.qmd -P SA
```
its unclear how this is done when defining the render using `yml`
```qmd
project: 
  type: website
  render:
    analysis.qmd
website
  title: "Test"
  navbar:
    left:
      - href: analysis.qmd
         text: USA analysis
      - href: analysis.qmd
         text: UK analysis
      - href: analysis.qmd
        text: South Africa analysis
format: 
  html:
    toc: true
```

Feels like there should be a render section in the documentation here: https://quarto.org/docs/projects/quarto-projects.html#render-targets

**Comment by f2il-kieranmace** [comment, marked as answer]
I've come up with a solution that works for me: 

This is my `_quarto.yml` file
```qmd
project: 
  type: website
  render:
    - index.qmd
    - US_analysis.qmd
    - UK_analysis.qmd
    - SA_analysis.qmd
website
  title: "Test"
  navbar:
    left:
      - href: US_analysis.qmd
         text: USA analysis
      - href: UK_analysis.qmd
         text: UK analysis
      - href: SA_analysis.qmd
        text: South Africa analysis
format: 
  html:
    toc: true
```

second step was to create a generic `analysis.qmd` file, which has all of the R code. 

final step is to create tiny `qmd` files for each page, that effectively are the params, and include the `analysis.qmd` code. here's an example: 

```qmd
---
title: "USA"
params:
  country: USA
---
{{< include analysis.qmd >}}
```

**Reply by cscheid** [reply]
This is probably the cleanest way to do it right now, especially if you need all the files to be present in a single website.

**Reply by mcanouil** [reply]
Indeed, to pass parameters to book/website projects, you would need something to associate parameters (eventually as YAML files) to each pages somehow.

I personally use multiple times a "template" in my work with a directory containing several YAML files with all the parameters.
This way I can compile several version of the document with a lot of parameters using for loops in bash and quarto CLI in it.

**Reply by aborruso** [reply]
@cscheid @mcanouil  I have a related question. If I have to open a new thread, tell me and I'll do it.

How to use parameter in front-matter?

Is there a way to set something similar to

````
---
title: My name is `{python} myPar`
---
````

and then run `quarto render -P myPar:Andrea` to get

````
---
title: My name is Andrea
---
````

Thank you

**Reply by mcanouil** [reply]
Why do you need/want parameters here? Parameters are for computations.

Simply use the `--metadate/-M` flag.

```sh
quarto render index.qmd --metadata title:"My Name is Whatever"
```

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/846da826-dd7d-4389-9883-77b1e39bc902">

If you don't like redundancy:

````qmd
---
title: "My Name is {{< meta custom-author >}}"
custom-author: Me
format: html
---

This is a playground for Quarto.
````

```sh
quarto render index.qmd --metadata custom-author:"Nobody"
```

**Reply by aborruso** [reply]
Thank you @mcanouil. It's 6 p.m. here, and my 50s are feeling strong now: I'm going to break off and go to sleep.

Sorry for the wrong and stupid question

**Reply by cscheid** [reply]
For what's worth (speaking as a Quarto developer here): I do think that it's a bit ugly that we have this separation between metadata and parameters. I wonder what it would take for us to improve things.

**Reply by mcanouil** [reply]
> For what's worth (speaking as a Quarto developer here): I do think that it's a bit ugly that we have this separation between metadata and parameters. I wonder what it would take for us to improve things.

Possibly something to be done when harmonising Jupyter/knitr engine handling of parameters.

**Reply by mcanouil** [reply]
> Sorry for the wrong and stupid question

The only wrong/stupid questions are the ones never asked.
